src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/endpoints/JurisdictionEndpoint.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.endpoints;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.NoUserConfiguredException;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Map.Entry;
+
+@Component
+@Endpoint(id = "jurisdictions")
+@EnableConfigurationProperties(JurisdictionToUserMapping.class)
+public class JurisdictionEndpoint {
+
+    private static final Logger log = LoggerFactory.getLogger(JurisdictionEndpoint.class);
+
+    private final Map<String, Credential> jurisdictions;
+    private final Function<String, Credential> credential;
```

Comment:
surely this can be clearer

Reply:
Please give more input as personally it is pretty clear

Reply:
Store `JurisdictionToUserMapping` and use it.
So instead of 
```
private final Map<String, Credential> jurisdictions;
private final Function<String, Credential> credentialProvider;
private final IdamClient idamClient;

public JurisdictionEndpoint(
    JurisdictionToUserMapping mapping,
    IdamClient idamClient
) {
    jurisdictions = mapping.getUsers();
    credentialProvider = mapping::getUser;
    this.idamClient = idamClient;
}
```
you can simply do
```
private JurisdictionToUserMapping jurisdictionToUserMapping;
private final IdamClient idamClient;

public JurisdictionEndpoint(
    JurisdictionToUserMapping mapping,
    IdamClient idamClient
) {
    this.jurisdictionToUserMapping = jurisdictionToUserMapping;
    this.idamClient = idamClient;
}
```